pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -646,6 +654,82 @@ func (r *Reconciler) fetchRunningInstancesInInstanceGroup(projectID string, zone
 	return instanceSets, nil
 }
 
+// registerNewInstanceGroup registers an instance group when there is an instance
+// that is using that unkown instance group. After the creation of the instance group
+// we add patch our internal load balancer with that new instance group.
+func (r *Reconciler) registerNewInstanceGroup() error {
+	_, err := r.computeService.InstanceGroupInsert(r.projectID, r.providerSpec.Zone, &compute.InstanceGroup{
+		Name:       r.controlPlaneGroupName(),
+		Region:     r.providerSpec.Region,
+		Zone:       r.providerSpec.Zone,
+		Network:    r.instanceGroupNetworkName(),
+		Subnetwork: r.instanceGroupSubNetworkName(),
+	})
+	if err != nil {
+		return fmt.Errorf("instanceGroupInsert request failed: %v", err)
+	}
+
+	return nil
+}
+
+func (r *Reconciler) updateBackendServiceWithInstanceGroup() error {
+	// We have to wait for the instance group to be ready
+	backoff := wait.Backoff{
+		Duration: 10 * time.Second,
+		Steps:    10,
+		Factor:   1.0,
+		Jitter:   1.0,
+	}
```

Comment:
If you return an error from the function, it causes the controller to requeue and do backoff anyway, we shouldn't bother with our own back off here as this makes this blocking, requeue, process other things, come back to it

Reply:
I experienced that, if I used `retry.DefaultBackoff` it didn't provide enough time for the instancegroup to become ready.

Reply:
Also, the issue is with the structure of the code. If we don't patch the backend service right after registering the new instance group, we never get to execute that code again.